orderbook/src/orderbook.rs

```diff
@@ -52,12 +53,21 @@ impl OrderBook {
 }
 
 struct InvalidSignatureError {}
-fn user_order_to_full_order(user_order: UserOrder) -> Result<Order, InvalidSignatureError> {
-    // TODO: verify signature and extract owner
+fn user_order_to_full_order(user_order: OrderCreation) -> Result<Order, InvalidSignatureError> {
+    // TODO: verify signature and extract owner, get orderDigest
+    let owner = H160::zero();
+    let digest = H256::zero();
+    let valid_to = user_order.valid_to.to_be_bytes();
+    let mut uid = [0 as u8; 56];
+    uid.copy_from_slice(&[digest.as_bytes(), owner.as_bytes(), &valid_to].concat());
```

Comment:
It should be possible to use the standard library `TryFrom` implementation to convert directly to an array:
```suggestion
    let uid = OrderUid([digest.as_bytes(), owner.as_bytes(), &valid_to].concat().try_into());
```

Reply:
Oh right, it was not working for me, just because of my old rust version. Nice thanks